aws/scenarios/microVMs/microVMs/libvirt_fs.go

```diff
@@ -0,0 +1,246 @@
+package microVMs
+
+import (
+	"fmt"
+
+	_ "embed"
+
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/microVMs/vmconfig"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const basefsName = "custom-fsbase"
+
+//go:embed resources/volume.xml
+var volumeXMLTemplate string
+
+//go:embed resources/pool.xml
+var poolXMLTemplate string
+
+type FSImage struct {
+	imageName     string
+	imagePath     string
+	imageSource   string
+	volumeKey     string
+	volumeXML     string
+	volumeXMLPath string
+}
+
+type libvirtFilesystem struct {
+	poolName string
+	poolXML  string
+	images   []FSImage
+}
+
+func generatePoolPath(name string) string {
+	return "/pool/" + name + "/"
+}
+
+func generateVolumeKey(pool string) string {
+	return generatePoolPath(pool) + basefsName
+}
+
+func getImagePath(name string) string {
+	return fmt.Sprintf("/tmp/%s", name)
+}
+
+func NewLibvirtFS(poolName string, img *vmconfig.Image) *libvirtFilesystem {
+	poolPath := generatePoolPath(poolName)
+	poolXML := fmt.Sprintf(poolXMLTemplate, poolName, poolPath)
+
+	volKey := generateVolumeKey(poolName)
+	return &libvirtFilesystem{
+		poolName: poolName,
+		poolXML:  poolXML,
+		images: []FSImage{
+			{
+				imageName:     img.ImageName,
+				imagePath:     getImagePath(img.ImageName),
+				imageSource:   img.ImageSourceURI,
+				volumeKey:     volKey,
+				volumeXML:     fmt.Sprintf(volumeXMLTemplate, basefsName, volKey, volKey),
+				volumeXMLPath: fmt.Sprintf("/tmp/volume-%s.xml", img.ImageName),
+			},
+		},
+	}
+}
+
+func downloadRootfs(fs *libvirtFilesystem, runner *command.Runner, depends []pulumi.Resource) ([]pulumi.Resource, error) {
```

Comment:
Considering the number of commands in this file, would be it clearer to simply copy (once) and execute a bash script?

Reply:
We plan to use this scenario for local testing as well. Using a single bash script could be problematic if the user wants to add and remove different VMs from the configuration, since we will have to rebuild the volumes for each every time the configuration file is edited.
On the other hand by having pulumi track each command, only the relevant commands are rerun on each modification of the microVM configuration file.

Reply:
Maybe you can generate the script on-the-fly from a template or something.
I think it makes sense if commands are slowish (few seconds) or not idempotent, otherwise I think it's okay to just call the script all the time

Reply:
We are planning to generate a custom ami for our purposes, and much of this logic will be moved there.
I can refactor this code in a followup PR once we have the relevant parts ready.